src/transformers/models/gpt2/modeling_gpt2.py

```diff
@@ -1061,11 +1190,6 @@ def forward(
         # head_mask has shape n_layer x batch x n_heads x N x N
         head_mask = self.get_head_mask(head_mask, self.config.n_layer)
 
-        if inputs_embeds is None:
```

Comment:
Why this has been removed ? This looks like a breaking change no?

Reply:
Ah ic, yea it looks weird via the github diff but if you look above it's added right back there (L1137-1141). I just changed the order of operations to fit in with the sdpa mask creation.